src/coreclr/vm/arm64/asmhelpers.S

```diff
@@ -23,6 +23,14 @@ LEAF_ENTRY GetDataCacheZeroIDReg, _TEXT
     ret     lr
 LEAF_END GetDataCacheZeroIDReg, _TEXT
 
+#if defined(_MSC_VER)
```

Comment:
Is this needed here and the same for the ifdef in the other file? I thought that `asmhelpers.S` was exclusively used by non-MSVC and `asmhelpers.asm` was exclusively used by MSVC (or rather MASM) and so there wasn't a need for such ifdefs (and why we don't have them elsewhere in these files)

Reply:
yes, I don't think `_MSC_VER` is needed.

Reply:
I know the call to this method is guarded under `if (SVE)`, I am actually wondering the implications of calling it by someone without having this check? Is there a way to full proof this to return 0 or something if ran on non-sve arm64 machines?

Reply:
Not really, Arm64 doesn't have a user-mode way of checking CPU feature support

Reply:
So basically, if someone calls this method unintentionally, on non-sve arm64, they would just hit "Internal error", which is a fair thing to be.

Reply:
Ooh, I didn't know which one was used by MSVC and non-MSVC. In this case, we need to remove the non-MSVC bits from .S file because we need the "sve" attribute to compile it.